### ISSUE TITLE
Doesn't work with most tools (.NET Assemblies?)

### ISSUE THREAD

**Issue by MexHigh** [cpu0x00/SharpReflectivePEInjection#1]
Some feedback on your tool with some of my tests.

### Working
- `.\SharpReflectivePEInjection.exe -file C:\Windows\System32\calc.exe -syscalls`
- `.\SharpReflectivePEInjection.exe -file C:\Windows\System32\calc.exe`

### Not working
- `.\SharpReflectivePEInjection.exe -file C:\Windows\System32\cmd.exe -a "/c calc.exe"` _(exits after patching args)_
- `.\SharpReflectivePEInjection.exe -file .\SharpHound.exe` _(Sharphound compiled to x64, exits after patching args)_
- `.\SharpReflectivePEInjection.exe -file .\SharpHound.exe -a "-c DCOnly"` _(Sharphound compiled to x64, exits after patching args)_
- `.\SharpReflectivePEInjection.exe -file .\SharpHound.exe -a "--help"` _(exits instantly after showing help page, because the content of the `-a` flag is recognized as flag for `SharpReflectivePEInjection.exe`)_

It seems like the tool does not work with .NET assemblies at all, even when compiled for x64 explicitly.

**Comment by cpu0x00** [comment]
@MexHigh yes its not intended  to load DotNet assemblies, its for loading unmanaged binaries like C/C++ Nim Rust so on, im thinking of adding a DotNet loading functionality but as for now its not intended for DotNet Assemblies

**Comment by MexHigh** [comment]
Ahh that makes sense. Thanks for the clarification.

Should I leave this open as a feautre request?

**Comment by cpu0x00** [comment]
@MexHigh yes its not inteded to load DotNet binaries, its for loading unmanaged binaries like C/C++ Nim Rust so on, im thinking of adding a DotNet loading functionality but as for now its not intended for DotNet Assemblies

**Comment by cpu0x00** [comment]
Sure, and feel free to add more feedback

**Comment by cpu0x00** [comment]
@MexHigh 
i tested the cmd.exe issue you had and it worked 
![test_cmd](https://github.com/cpu0x00/SharpReflectivePEInjection/assets/86830248/bd8f0e9c-eb8a-4907-abdb-56a08be71373)

if you can give me more detail regards the issue or maybe try to rerun it and update me on it

**Comment by cpu0x00** [comment]
![2023-11-17_03-45](https://github.com/cpu0x00/SharpReflectivePEInjection/assets/86830248/45d79290-8e75-45be-af41-4b552c7614ba)

**Comment by Justice-Wayne** [comment]
Hello author,
   Very nice project, I have a similar one. But I didn't solve the problem of passing parameters. I tested this project and it is not compatible with golang. There will be some weird situations.



<img width="957" alt="image" src="https://github.com/cpu0x00/SharpReflectivePEInjection/assets/88639842/86f24666-f25b-4bb6-b627-079acdcb789c">


<img width="998" alt="image" src="https://github.com/cpu0x00/SharpReflectivePEInjection/assets/88639842/9c4e25bc-94ff-4c1c-a8e7-151938868bea">

**Comment by cpu0x00** [comment]
@Justice-Wayne  i haven't tested it with golang only with C/C++, can you provide the code for your golang test binary

**Comment by MexHigh** [comment]
> @MexHigh i tested the cmd.exe issue you had and it worked ![test_cmd](https://user-images.githubusercontent.com/86830248/283643474-bd8f0e9c-eb8a-4907-abdb-56a08be71373.png)
> 
> if you can give me more detail regards the issue or maybe try to rerun it and update me on it

Does not work for me, for both calc.exe and whoami. Here's what I see:

![image](https://github.com/cpu0x00/SharpReflectivePEInjection/assets/50419884/aa4a00e5-0724-44e7-b6ef-35fe0cc931ed)

Some system information:
-  Windows 10 22H2 (19045.3693)
- Domain joined (fully on-premise)
- Defender deactivated, no Defender for Endpoint

**Comment by Justice-Wayne** [comment]
> i haven't tested it with golang only with C/C++, can you provide the code for your golang test binary

@cpu0x00  thank your answer ， 
https://github.com/moonD4rk/HackBrowserData 
https://github.com/fatedier/frp
https://github.com/shadow1ng/fscan

**Comment by cpu0x00** [comment]
@MexHigh are you sure you are compiling right?

**Comment by MexHigh** [comment]
> @MexHigh are you sure you are compiling right?

I wasn't able to find any compilation instructions so I just compiled the provided Release > Any CPU configuration with Visual Studio 2022. Do I need to target a specific architecture?

EDIT: With a binary directly targeted to x64 the same happens (no error, just exit after patching args, nothing else happens). With a x86 binary, i see another exception:

![image](https://github.com/cpu0x00/SharpReflectivePEInjection/assets/50419884/9fd2bf98-ecde-40d2-aca2-7ab48fd9d731)

**Comment by cpu0x00** [comment]
When compiled for x64 can only load for x64 and vice versa

The tool support memory alignment for both architectures but it needs to be compiled for the specific on its gonna be used in

**Comment by cpu0x00** [comment]
@MexHigh hi,

turns out you were right it didn't map x32 , just fixed it, thank you for the testing
